### PR TITLE
Control Stack cleanup and revised debug data collection message

### PR DESCRIPTION
**Pull request by menson168**

### CODE REVIEW

**Comment by menson168** [comment]
#104

**Comment by drofp** [comment]
@menson168 Please make an issue and corresponding branch for future edits

File: src/modules/interface/msg/Trajectory.msg [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Header header
+
+# Trajectory Message that contains both position of the vehicle and linear velocity
+
+interface/Waypoint[] path
```
Comment (drofp) [inline src/modules/interface/msg/Trajectory.msg:5]: Is this intended to replace the Path message published on `/planning/trajectory` by the linear path interpolation node?

File: src/modules/interface/msg/Waypoint.msg [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Header header
+
+# Waypoint Message that contains both position of the vehicle and linear velocity of a point
+
+# Should only contain position.x/y
+geometry_msgs/PoseStamped poses
```
Comment (drofp) [inline src/modules/interface/msg/Waypoint.msg]: This message type is for a single pose, not an array of poses
```suggestion
geometry_msgs/PoseStamped pose
```

Comment (drofp) [reply src/modules/interface/msg/Waypoint.msg]: are you looking for this message instead? http://docs.ros.org/kinetic/api/geometry_msgs/html/msg/PoseArray.html

File: src/modules/interface/msg/Trajectory.msg (hunk 1 above)
Comment (drofp) [reply src/modules/interface/msg/Trajectory.msg:5]: This message type seems appropriate for recording "trajectory":
http://docs.ros.org/kinetic/api/geometry_msgs/html/msg/Vector3.html

Comment (menson168) [reply src/modules/interface/msg/Trajectory.msg:5]: Vector 3 describes a vector, not a position. The implementation of trajectory messages is to include velocity requirement in each waypoint. Therefore, I declared a waypoint message which contains the pose and velocity for each waypoint, then have the trajectory message to be an array of these waypoints.

File: src/modules/interface/msg/Waypoint.msg (hunk 2 above)
Comment (menson168) [reply src/modules/interface/msg/Waypoint.msg]: Referring to the comment above. Similar to the definition of nav_msgs/Path, we will be using an array definition of waypoints, which is a custom type of waypoint definition from us.

**Review by drofp — APPROVED** [review]

**Pull request merged**